File: APPTOPI/app/src/main/java/io/topi/apptopi/adapter/GitItemAdapter.java
package io.topi.apptopi.adapter;

import android.content.Context;
import android.content.Intent;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.request.RequestOptions;

import org.androidannotations.annotations.EViewGroup;
import org.androidannotations.annotations.ViewById;

import io.topi.apptopi.R;
import io.topi.apptopi.activity.MainActivity;
import io.topi.apptopi.activity.MoreDetailsActivity;
import io.topi.apptopi.activity.MoreDetailsActivity_;
import io.topi.apptopi.model.GitItem;

@EViewGroup(R.layout.cell_git)
public class GitItemAdapter extends RelativeLayout {

    @ViewById
    protected TextView tvNameRepo;
    @ViewById
    protected TextView tvDescriptionRepo;
    @ViewById
    protected TextView tvForks;
    @ViewById
    protected TextView tvStargazers;
    @ViewById
    protected ImageView ivUser;
    @ViewById
    protected TextView tvUsernameUser;
    @ViewById
    protected TextView tvMoreDetails;

    public GitItemAdapter(Context context){
        super(context);
    }

    public void bind(GitItem gitItem){
        tvNameRepo.setText(gitItem.getName());
        int limitDescription = 70;
        String description = "";
        for(int d=0; d<gitItem.getDescription().length(); d++){
            if(d<limitDescription){
                description += (gitItem.getDescription().charAt(d) + "");
            }
        }
        if(gitItem.getDescription().length() > description.length()){
            description += "...";
        }
        tvDescriptionRepo.setText(description);
        tvForks.setText(Integer.toString(gitItem.getForks()));
        tvStargazers.setText(Integer.toString(gitItem.getStargazers_count()));

        if(gitItem.getOwner() != null) {
            if(gitItem.getOwner().getAvatar_url() != null && gitItem.getOwner().getAvatar_url().length() > 0) {
                RequestOptions requestOptions = new RequestOptions()
                        .placeholder(R.drawable.person_circle)
                        .dontAnimate().centerCrop()
                        .diskCacheStrategy(DiskCacheStrategy.ALL);
                Glide.with(getContext()).load(gitItem.getOwner().getAvatar_url())
                        .apply(requestOptions).into(ivUser);
            }
            tvUsernameUser.setText(gitItem.getOwner().getLogin());
        }
        tvMoreDetails.setOnClickListener(v -> {
            MoreDetailsActivity_.intent(getContext()).extra("git_bundle", gitItem).start();
        });
    }
}

File: APPTOPI/app/src/main/java/io/topi/apptopi/main/view/adapter/GitItemAdapter.kt
package io.topi.apptopi.main.view.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.request.RequestOptions
import io.topi.apptopi.R
import io.topi.apptopi.main.model.GitItem
import io.topi.apptopi.main.view.activity.MoreDetailsActivity
import kotlinx.android.synthetic.main.cell_git.view.*
import kotlinx.android.synthetic.main.layout_fork_side_star.view.*

class GitItemAdapter: RecyclerView.Adapter<GitItemAdapter.ViewHolder>() {

    private var itens: ArrayList<GitItem> = arrayListOf()

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val gitItem = itens.get(position)
        holder.setIsRecyclable(false)
        holder.tvNameRepo.text = gitItem.name
        holder.tvDescriptionRepo.text = if(gitItem.description != null)
            "${if(gitItem.description.length > 70) gitItem.description.substring(0, 70) else gitItem.description }" +
                    "${if (gitItem.description.length > 70)"..." else ""}" else ""
        holder.tvForks.text = gitItem.forks.toString()
        holder.tvStargazers.text = gitItem.stargazers_count.toString()

        if(gitItem.owner.avatar_url.isNotEmpty()) {
            val requestOptions: RequestOptions = RequestOptions().placeholder(R.drawable.person_circle)
                    .dontAnimate().centerCrop().diskCacheStrategy(DiskCacheStrategy.ALL)
            Glide.with(holder.itemView.context).load(gitItem.owner.avatar_url)
                    .apply(requestOptions).into(holder.ivUser)
        }
        holder.tvUsernameUser.text = gitItem.owner.login

        holder.tvMoreDetails.setOnClickListener {view ->
            MoreDetailsActivity.showActivity(holder.itemView.context, gitItem)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.cell_git, parent, false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return itens.size
    }

    fun updateAdapter(gitItems: ArrayList<GitItem>){
        this.itens = gitItems;
        notifyDataSetChanged();
    }

    class ViewHolder(itemView: View): RecyclerView.ViewHolder(itemView){
        val tvNameRepo = itemView.tvNameRepo
        val tvDescriptionRepo = itemView.tvDescriptionRepo
        val tvForks = itemView.tvForks
        val tvStargazers = itemView.tvStargazers
        val ivUser = itemView.ivUser
        val tvUsernameUser = itemView.tvUsernameUser
        val tvMoreDetails = itemView.tvMoreDetails
    }
}